image_proc/src/image_proc.cpp

```diff
@@ -89,19 +75,19 @@ int main(int argc, char * argv[])
 
   // Debayer nodelet, image_raw -> image_mono, image_color
   auto debayer_node = std::make_shared<image_proc::DebayerNode>(options);
-  debayer_node->declare_parameter("camera_namespace", camera_namespace);
+  debayer_node->set_parameter(rclcpp::Parameter("camera_namespace", camera_namespace));
```

Comment:
Its not clear to me what your refactor buys us vs the old changes of declaring the parameters outside the node itself and here in the main()

Reply:
I'm not sure how one would do it if we want to use it as a component. Before, trying to use it as a component, the callback was never triggered and the camera_namespace would not be declared.

Reply:
mhm. So I'm trying to rectify 2 things, both the use in the lanuch file that you have so you can do commandline remapping and the use of the nodes as individual components. Right now if you don't use this main() for the library, you really cant use the components themselves because the declares don't set logical defaults and they don't try to get the parameter themselves. 

You can set arguments through the NodeOptions class http://docs.ros2.org/dashing/api/rclcpp/classrclcpp_1_1NodeOptions.html#ab0c61163fb466ceb980a001bdce2fb7a